pkg/nuclide-clang-rpc/lib/ClangFlagsManager.js

```diff
@@ -266,6 +304,34 @@ export default class ClangFlagsManager {
     return null;
   }
 
+  async _loadProjectCompilerFlags(flagsFile: string): Promise<?ClangProjectFlags> {
+    if (this._clangProjectFlags != null) {
+      return this._clangProjectFlags;
+    }
+
+    let result = null;
+    try {
+      const contents = await fsPromise.readFile(flagsFile, 'utf8');
+      const data = JSON.parse(contents);
+      invariant(data instanceof Object);
+      const {copts, nocopts, includes} = data;
+      const extraCompilerFlags = {flags: [], rawData: null, flagsFile};
```

Comment:
As mentioned above, I'd just leave these as the raw arrays.
Also it's probably a good idea to make these fields optional, so make sure you handle the case where they're null.